screenpy/actions/silently.py

```diff
@@ -0,0 +1,126 @@
+"""
+Silently allows for "disabling" logging on successful actions & tests
+"""
+from __future__ import annotations
+
+from typing import Any, TypeVar, Union, overload
+
+from hamcrest.core.base_matcher import Matcher
+from typing_extensions import TypeAlias
+
+from screenpy import settings
+from screenpy.actor import Actor
+from screenpy.exceptions import NotAnswerable, NotPerformable, NotResolvable
+from screenpy.pacing import the_narrator
+from screenpy.protocols import Answerable, Performable, Resolvable
+
+T = TypeVar("T")
+
+
+class SilentlyMixin:
+    """
+    Silently needs to mimic the ducks (i.e. objects) they are wrapping.
+    All of the attributes from the "duck" should be exposed by the Silently object.
+    """
+
+    def __getattr__(self, key: Any) -> Any:
+        try:
+            return getattr(self.duck, key)
+        except AttributeError as exc:
+            msg = (
+                f"{self.__class__.__name__}({self.duck.__class__.__name__}) "
+                f"has no attribute '{key}'"
+            )
+            raise AttributeError(msg) from exc
+
+
+class SilentlyPerformable(Performable, SilentlyMixin):
+    """Act like the Performable passed in but kink the cable when performing"""
+
+    def perform_as(self, actor: Actor) -> None:
+        with the_narrator.mic_cable_kinked():
+            self.duck.perform_as(actor)
+            if not settings.UNABRIDGED_LOGGING:
+                the_narrator.clear_backup()
+            return
+
+    def __init__(self, duck: Performable):
+        if not isinstance(duck, Performable):
+            raise NotPerformable(
+                "SilentlyPerformable only works with Performable. "
+                "Use `Silently` instead."
+            )
+        self.duck = duck
+
+
+class SilentlyAnswerable(Answerable, SilentlyMixin):
+    """Act like the Answerable passed in but kink the cable when answering"""
+
+    def answered_by(self, actor: Actor) -> Any:
+        with the_narrator.mic_cable_kinked():
+            thing = self.duck.answered_by(actor)
+            if not settings.UNABRIDGED_LOGGING:
+                the_narrator.clear_backup()
+            return thing
+
+    def __init__(self, duck: Answerable):
+        if not isinstance(duck, Answerable):
+            raise NotAnswerable(
+                "SilentlyAnswerable only works with Answerable. Use `Silently` instead."
+            )
+        self.duck = duck
+
+
+class SilentlyResolvable(Resolvable, SilentlyMixin):
+    """Act like the Resolvable passed in but kink the cable when resolving"""
+
```

Comment:
These are TERRRIBLE descriptions.  I need to change them.

Reply:
How about:

```python
class SilentlyPerformable:
    """-Silently performant-"""
    
class SilentlyAnswerable:
    """-Silently answered-"""

class SilentlyResolvable:
    """-Silently resolute-"""
```